src/main/resources/db/migration/R__functions.sql

```diff
@@ -27,13 +27,58 @@ CREATE OR REPLACE FUNCTION merge_object(bytes_ BYTEA,
                                         url_ TEXT,
                                         client_id_ TEXT) RETURNS SETOF objects AS
 $$
-    INSERT INTO objects (hash, content, url, client_id)
-         VALUES (hash_, bytes_, url_, client_id_)
-    ON CONFLICT (hash) DO UPDATE
-            SET deleted_at = NULL
+    INSERT INTO objects (url, hash, content, client_id)
+         VALUES (url_, hash_, bytes_, client_id_)
+    ON CONFLICT (url, hash) DO UPDATE
+            SET deleted_at = NULL,
+                client_id = EXCLUDED.client_id
```

Comment:
In practice I don't think we'll have this, since we'll have to verify that a client does not publish objects for URLs that are not inside their namespace. Right now it is a problem, since we do not check for this.

I'd actually prefer not to "reuse" rows this way, but always insert a new row for every new object, even when the hash was already seen before.